paperqa/agents/env.py

```diff
@@ -157,10 +157,26 @@ async def reset(self) -> tuple[list[Message], list[Tool]]:
     def export_frame(self) -> Frame:
         return Frame(state=self.state, info={"query": self._query})
 
+    USE_POST_PROCESSED_REWARD: ClassVar[float] = 0.0
+
     async def step(
         self, action: ToolRequestMessage
     ) -> tuple[list[Message], float, bool, bool]:
         self.state.session.add_tokens(action)  # Add usage for action if present
+        if not action.tool_calls:
+            return (
+                [
+                    Message(
+                        content=(
+                            f"{GenerateAnswer.FAILED_TO_ANSWER} Agent specified 0 tool"
+                            " calls, which means done."
```

Comment:
Should we say "... which means unsure."?

Reply:
Yes this is a great point. I need to really think about why we have `GenerateAnswer.FAILED_TO_ANSWER`, it seems like it shouldn't exist.

I will circle back on this tomorrow after some sleep

Reply:
Okay I updated this to be:

```python
return (
    # NOTE: don't put:
    # - GenerateAnswer.FAILED_TO_ANSWER here because this wasn't a failure
    # - 'cannot answer' because that information belongs in
    #   PQASession.answer, not in the message history
    # Let's just put a nice message about being done :)
    [Message(content="Agent specified 0 tool calls, which means done.")],
    ...
)
```

Hopefully that makes sense